Sources/Filters/Core/CleanPolyData/index.js

```diff
@@ -0,0 +1,485 @@
+        pointMap[ptId] = numUsedPts.value++;
+        newPts.setPoint(pointMap[ptId], newx);
+        outputPD.passData(inputPD, ptId, pointMap[ptId]);
+      }
+      return pointMap[ptId];
+    }
+    const newPtId = model.locator.insertUniquePoint(newx).id;
+    if (!model.copiedPoints.has(newPtId)) {
+      model.copiedPoints.add(newPtId);
+      outputPD.passData(inputPD, ptId, newPtId);
+    }
+    return newPtId;
+  }
+
+  publicAPI.operateOnPoint = (inPt, outPt) => {
+    outPt[0] = inPt[0];
+    outPt[1] = inPt[1];
+    outPt[2] = inPt[2];
+  };
+
+  publicAPI.operateOnBounds = (inBounds, outBounds) => {
+    for (let i = 0; i < 6; i++) {
+      outBounds[i] = inBounds[i];
+    }
+  };
+
+  publicAPI.createDefaultLocator = (input) => {
+    let tol;
+    if (model.toleranceIsAbsolute) {
+      tol = model.absoluteTolerance;
+    } else {
+      tol = input ? model.tolerance * input.getLength() : model.tolerance;
+    }
+
+    if (!model.locator) {
+      if (tol === 0.0) {
+        model.locator = vtkMergePoints.newInstance();
+      } else {
+        model.locator = vtkPointLocator.newInstance();
+      }
+    }
+    // Check if tolerance changed from zero to non-zero
+    else if (tol > 0.0 && model.locator.getTolerance() === 0.0) {
+      model.locator = vtkPointLocator.newInstance();
+    }
+  };
+
+  publicAPI.requestData = (inData, outData) => {
+    const input = inData[0];
+    const output = outData[0]?.initialize() || vtkPolyData.newInstance();
+    outData[0] = output;
+
+    const inPts = input.getPoints();
+    const numPts = input.getNumberOfPoints();
+
+    if (!inPts || numPts < 1) {
+      outData[0] = output;
+      return;
+    }
+
+    const updatedPts = new Array(input.getMaxCellSize());
+    const numUsedPts = { value: 0 };
+
+    const precision = model.outputPointsPrecision;
+    let pointType = inPts.getDataType();
+    if (precision) {
+      pointType =
+        precision === DesiredOutputPrecision.DOUBLE
+          ? VtkDataTypes.DOUBLE
+          : VtkDataTypes.FLOAT;
+    }
+    const newPts = vtkPoints.newInstance({ dataType: pointType });
+    const inVerts = input.getVerts();
+    const inLines = input.getLines();
+    const inPolys = input.getPolys();
+    const inStrips = input.getStrips();
+
+    let newVerts = null;
+    let newLines = null;
+    let newPolys = null;
+    let newStrips = null;
+
+    const inputPD = input.getPointData();
+    const inputCD = input.getCellData();
+    const outputPD = output.getPointData();
+    const outputCD = output.getCellData();
+
+    let pointMap = null;
+    if (model.pointMerging) {
+      publicAPI.createDefaultLocator(input);
+
+      if (model.toleranceIsAbsolute) {
+        model.locator.setTolerance(model.absoluteTolerance);
+      } else {
+        model.locator.setTolerance(model.tolerance * input.getLength());
+      }
+
+      const originalBounds = input.getBounds();
+      const mappedBounds = [...originalBounds];
```

Comment:
no need to initialize to originalBounds, because you will do it line 134.
`const mappedBounds = [];`